enacts/calc.py

```diff
@@ -65,28 +74,30 @@ def water_balance(
     Examples
     --------
     """
-    # Get time_coord info
-    time_coord_size = daily_rain[time_coord].size
-    # Get all the rain-et deltas:
-    delta_rain_et = daily_rain - et
-    # Intializing sm
-    soil_moisture = xr.DataArray(
-        data=np.empty(daily_rain.shape),
-        dims=daily_rain.dims,
-        coords=daily_rain.coords,
-        name="soil_moisture",
-        attrs=dict(description="Soil Moisture", units="mm"),
+    # Initializing soil_moisture
+    soil_moisture = water_balance_step(
+        sminit,
+        daily_rain.isel({time_dim: 0}).expand_dims(dim=time_dim),
```

Comment:
yes because it conveniently assigns the right 1-point time dimension to first iteration of soil_moisture, which I either want to use with align if not reduce, or have the time dim information if reduce.